### PR TITLE
feat: headings are now links and can be copied upon click

### PR DESCRIPTION
**Pull request by barrybtw**
Closes #<issue>

## ✅ Checklist

- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] I performed a functional test on my final commit

---

## Changelog

Headings inside the docs layout are now links, and upon click will copy to the clickboard.
---

💯

### CODE REVIEW

File: .changeset/dry-eagles-travel.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@ct3a/www": minor
+---
+
+feat: headings are now links and can be copied upon click
```
Comment (juliusmarminge) [inline .changeset/dry-eagles-travel.md:5]: Remove this. We dont publish the www package so no changesets are needed

File: www/src/layouts/docs.astro [hunk 2]
```diff
@@ -39,6 +39,37 @@ const githubEditUrl = `${CONFIG.GITHUB_EDIT_URL}/${currentFile}`;
...
+          section.querySelectorAll("h2"),
+          section.querySelectorAll("h3"),
+        ];
+
+        const headings = [...headings1, ...headings2, ...headings3];
+
+        headings.forEach((heading) => {
+          const link = document.createElement("a");
+          link.href = `#${heading.id}`;
+          link.className = "heading-link";
```
Comment (juliusmarminge) [inline www/src/layouts/docs.astro:58]: Maybe change the stylings a bit here so they arent so " in your face".

Perhaps this is enough

```suggestion
          link.className = "hover:heading-link";
```

File: www/src/layouts/docs.astro [hunk 3]
```diff
@@ -39,6 +39,37 @@ const githubEditUrl = `${CONFIG.GITHUB_EDIT_URL}/${currentFile}`;
...
+          heading.innerHTML = "";
+          heading.appendChild(link);
+
+          heading.addEventListener("click", () => {
+            navigator.clipboard.writeText(
+              `${window.location.href}#${heading.id}`,
+            );
+          });
+
+          console.log(heading);
```
Comment (juliusmarminge) [inline www/src/layouts/docs.astro:69]:
```suggestion
```

**Review by juliusmarminge — COMMENTED** [review]
Nice!

File: www/src/layouts/docs.astro (hunk 2 above)
Comment (juliusmarminge) [reply www/src/layouts/docs.astro]: It didn't

File: www/src/layouts/docs.astro [hunk 4]
```diff
@@ -39,6 +39,37 @@ const githubEditUrl = `${CONFIG.GITHUB_EDIT_URL}/${currentFile}`;
...
+      function main() {
+        const section = document.querySelector("section.w-full");
+
+        const [headings1, headings2, headings3] = [
+          section.querySelectorAll("h1"),
+          section.querySelectorAll("h2"),
+          section.querySelectorAll("h3"),
+        ];
+
+        const headings = [...headings1, ...headings2, ...headings3];
```
Comment (juliusmarminge) [inline www/src/layouts/docs.astro:53]:
```suggestion
        const headings = section.querySelectorAll("h1, h2, h3, h4, h5");
```

Can merge all this into a single query

File: www/src/layouts/docs.astro (hunk 2 above)
Comment (barrybtw) [reply www/src/layouts/docs.astro]: [96cd482](https://github.com/t3-oss/create-t3-app/pull/639/commits/96cd4821c4da3a3243dcf998fa4a190557e8b0a6)

**Review by juliusmarminge — APPROVED** [review]
Looks and works great

**Pull request merged**